mozci/task.py

```diff
@@ -34,46 +35,37 @@ class Status(Enum):
     INTERMITTENT = 2
 
 
-SUITES = (
-    "mochitest-plain-gpu",
-    "mochitest-plain",
-    "mochitest-chrome-gpu",
-    "mochitest-chrome",
-    "mochitest-devtools-chrome",
-    "mochitest-browser-a11y",
-    "mochitest-browser-chrome",
-    "mochitest-browser-media",
-    "web-platform-tests-crashtest",
-    "web-platform-tests-reftest",
-    "web-platform-tests-wdspec",
-    "web-platform-tests-print-reftest",
-    "web-platform-tests",
-    "mochitest-media",
-    "mochitest-webgpu",
-    "mochitest-webgl1-ext",
-    "mochitest-webgl2-ext",
-    "mochitest-webgl1-core",
-    "mochitest-webgl2-core",
-    "mochitest-remote",
-    "mochitest-a11y",
-    "xpcshell",
-    "crashtest",
-    "jsreftest",
-    "reftest-no-accel",
-    "gtest",
-    "telemetry-tests-client",
-    "browser-screenshots",
-    "marionette-gpu",
-    "marionette",
-    "cppunit",
-    "firefox-ui-functional-remote",
-    "firefox-ui-functional-local",
-    "reftest",
-    "junit",
-    "test-verify",
-    "test-coverage",
-    "jittest",
-)
+class Suites:
+    SUITES: set[str] = set()
+
+    def __init__(self):
+        # code used to populate SUITE
+        try:
+            r = requests.get(
+                "https://hg.mozilla.org/mozilla-central/raw-file/tip/taskcluster/ci/test/test-sets.yml"
+            )
+            y = yaml.load(r.text, yaml.BaseLoader)
+        except requests.ConnectionError:
+            y = {}
+            pass
+        test_types = set()
+        for item in y:
+            test_types |= set(y[item])
+
+        # remove perf suites
+        self.SUITES = [
+            x
+            for x in test_types
+            if not (
+                x.startswith("talos")
+                or x.startswith("browsertime")
+                or x.startswith("awsy")
+            )
+        ]
+        self.SUITES.sort(reverse=True)
```

Comment:
Do we need to sort them?

Reply:
yes, so something like `reftest-gpu` is matched before `reftest`